jwql/utils/instrument_properties.py

```diff
@@ -0,0 +1,228 @@
+                                  "is {}. 1-amp frametime is {}. Reported frametime is {}.")
+                                 .format(amp4_time, amp1_time, frame_time))
+
+    if omit_reference_pixels:
+        # If requested, ignore reference pixels by adjusting the indexes of
+        # the amp boundaries.
+        with fits.open(filename) as hdu:
+            data_quality = hdu['PIXELDQ'].data
+
+        # Reference pixels should be flagged in the DQ array with the
+        # REFERENCE_PIXEL flag. Find the science pixels by looping for
+        # pixels that don't have that bit set.
+        scipix = np.where(data_quality & dqflags.pixel['REFERENCE_PIXEL'] == 0)
+        ymin = np.min(scipix[0])
+        xmin = np.min(scipix[1])
+        ymax = np.max(scipix[0]) + 1
+        xmax = np.max(scipix[1]) + 1
+
+        # Adjust the minimum and maximum x and y values if they are within
+        # the reference pixels
+        for key in amp_bounds:
+            bounds = amp_bounds[key]
+            prev_xmin, prev_ymin = bounds[0]
+            prev_xmax, prev_ymax = bounds[1]
+            if prev_xmin < xmin:
+                new_xmin = xmin
+            else:
+                new_xmin = prev_xmin
+            if prev_ymin < ymin:
+                new_ymin = ymin
+            else:
+                new_ymin = prev_ymin
+            if prev_xmax > xmax:
+                new_xmax = xmax
+            else:
+                new_xmax = prev_xmax
+            if prev_ymax > ymax:
+                new_ymax = ymax
+            else:
+                new_ymax = prev_ymax
+            amp_bounds[key] = [(new_xmin, new_ymin), (new_xmax, new_ymax)]
+    return num_amps, amp_bounds
+
+
+def calc_frame_time(instrument, aperture, xdim, ydim, amps, sample_time=1.e-5):
+    """Calculate the readout time for a single frame
+    of a given size and number of amplifiers. Note that for
+    NIRISS and FGS, the fast readout direction is opposite to
+    that in NIRCam, so we switch xdim and ydim so that we can
+    keep a single equation.
+
+    Parameters:
+    -----------
+    instrument : str
+        Name of the instrument being simulated
+
+    aperture : str
+        Name of aperture being simulated (e.g "NRCA1_FULL")
+        Currently this is only used to check for the FGS
+        ACQ1 aperture, which uses a unique value of colpad
+        below.
+
+    xdim : int
+        Number of columns in the frame
+
+    ydim : int
+        Number of rows in the frame
+
+    amps : int
+        Number of amplifiers used to read out the frame
+
+    sample_time : float
+        Time to sample a pixel, in seconds. For NIRCam/NIRISS/FGS
+        this is 10 microseconds = 1e-5 seconds
+
+    Returns:
+    --------
+    frametime : float
+        Readout time in seconds for the frame
+    """
+    instrument = instrument.lower()
+    if instrument == "nircam":
+        xs = xdim
```

Comment:
@bhilbert4 Not entirely sure what `xs` and `ys` are supposed to represent here.  Perhaps these variable names could be a bit more descriptive?

Reply:
Pulled out `xs` and `ys` in favor or using `xdim` and `ydim`.